### PR TITLE
Remove `--no-conda`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Remove `--no-conda`.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

`--no-conda` flag is no longer available.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/R/project-run.R [hunk 1]
```diff
@@ -42,7 +41,7 @@
 #' @export
 mlflow_run <- function(uri = ".", entry_point = NULL, version = NULL, parameters = NULL,
                        experiment_id = NULL, experiment_name = NULL, backend = NULL, backend_config = NULL,
-                       no_conda = FALSE, storage_dir = NULL) {
+                       env_manager = NULL, storage_dir = NULL) {
```
Comment (dbczumar) [inline mlflow/R/mlflow/R/project-run.R:44]: Nice! Separately, can we file a ticket to track removing conda from the R client install_mlflow() procedure entirely? I don't think there's a good reason to use conda just to install the MLflow Python client, and conda is problematic from a licensing perspective.

Comment (dbczumar) [reply mlflow/R/mlflow/R/project-run.R:44]: For reference, see https://github.com/mlflow/mlflow/issues/6439

**Review by dbczumar — APPROVED** [review]
LGTM!

File: mlflow/R/mlflow/R/project-run.R (hunk 1 above)
Comment (harupy) [reply mlflow/R/mlflow/R/project-run.R:44]: Got it 👍

Comment (harupy) [reply mlflow/R/mlflow/R/project-run.R:44]: Done. ML-24388.

File: mlflow/models/cli.py [hunk 2]
```diff
@@ -30,7 +30,6 @@ def commands():
 @cli_args.HOST
 @cli_args.TIMEOUT
 @cli_args.WORKERS
-@cli_args.NO_CONDA
 @cli_args.ENV_MANAGER
```
Comment (WeichenXu123) [inline mlflow/models/cli.py:33]: shall we change the default value of `ENV_MANAGER` option to be "virtualenv" ?

Comment (harupy) [reply mlflow/models/cli.py:33]: We did in https://github.com/mlflow/mlflow/pull/6501.

**Pull request merged**